### ISSUE TITLE
[ENHANCEMENT] Ability to add tags to jails

### ISSUE THREAD

**Issue by andy-isoc** [BastilleBSD/bastille#576]
In our use case, we are creating and deleting jails in distinct groups. Without an external database, we have no way of knowing which jails belong to which groups.

It would be ideal for us if there was the ability to add arbitrary text strings as "tags" when creating a jail. If it were possible to view these tags via `bastille list` (or better still to be able to query for jails with that tag via the same command), it would make it much easier for us to manage these groups of jails.

**Comment by bsdlme** [comment]
```

# bastille tags
Usage: bastille tags TARGET add tag1,tag2,...
       bastille tags TARGET delete tag1,tag2,...
       bastille tags TARGET list
Example: bastille tags JAILNAME add database,mysql
         bastille tags JAILNAME delete mysql
# bastille tags nicegui add python
# bastille tags nicegui add www,app,simple,nice
# bastille tags nicegui list
python
www
app
simple
nice
# bastille tags nicegui delete app,nice
# bastille tags nicegui list
python
www
simple

```

**Comment by bsdlme** [comment]
The tags are saved in the jail's base directory, where also rdr.conf, fstab, etc. reside.

**Comment by tobiastom** [comment]
How would I list all jails for a given tag?

**Comment by andy-isoc** [comment]
This would work fine for us, thanks for getting on to this so quickly!

As per @tobiastom question, it would be preferable if there was a way of listing all jails for a given tag. Perhaps that could be via `bastille list -a` ?

**Comment by bsdlme** [comment]
I re-worked it a bit. _list_ now shows the jail name with space separated tags on the same line. And searching is now possible:

```
# bastille tags ALL list
nicegui: nice www
niceproxy: caddy test www
# bastille tags ALL search www
nicegui
niceproxy
# bastille tags ALL search caddy
niceproxy
# bastille tags ALL search nice
nicegui
# bastille tags ALL delete nice
# bastille tags ALL search nice
#
```

**Comment by bsdlme** [comment]
> This would work fine for us, thanks for getting on to this so quickly!
> 
> As per @tobiastom question, it would be preferable if there was a way of listing all jails for a given tag. Perhaps that could be via `bastille list -a` ?

`bastille list -a` is theoretically good for this, but its output can be very unreadable already now.

**Comment by tobiastom** [comment]
@bsdlme I really like it. Looks really good. 

One idea to keep things simple:

```
# bastille tags ALL list
nicegui: nice www
niceproxy: caddy test www

# bastille tags ALL list www
nicegui
```

This would not introduce a new search sub-command and if list has a tag given it would just display the hosts, one per line. That said, maybe I'm missing some conflict with other commands here. :)

**Comment by bsdlme** [comment]
@tobiastom Good point! I updated the PR. Now it works like your suggested.

**Comment by andy-isoc** [comment]
Oh wow, this looks amazing and just what we need!

**Comment by andy-isoc** [comment]
I couldn't see a release policy listed anywhere. Do you know when this is likely to be released?

**Comment by tobiastom** [comment]
Based on the past @cedwards comes around sometimes and releases a new version. 😜
Not sure if there is a real policy.

**Comment by andy-isoc** [comment]
@cedwards do you have any idea when this might be released? We're trying to work out for our project whether we code a workaround, build from source or whether it's worth waiting. Thanks!

**Comment by jpds** [comment]
@andy-isoc Not a project maintainer but: just build from source, the patch is minimal and any tech debt incurred will be gone once you upgrade the package later.

**Comment by yaazkal** [comment]
> I couldn't see a release policy listed anywhere. Do you know when this is likely to be released?

well, no release policy I guess. But I can see this has been merged actually, so you can just clone the repo and run `make install`, then `git pull` and install it again whenever you new an updated version.

The main branch is usually stable enough.

**Comment by sveeke** [comment]
This is exactly what I need haha. I came here to make a issue for it with some suggestions, but then immediately found this issue. The PR looks great. @cedwards any idea when this will be merged/released? For now I can make it myself, but this is a great feature for the packaged versions :).

**Comment by yaazkal** [comment]
@sveeke this has already landed in version [`0.10.20230714`](https://github.com/BastilleBSD/bastille/releases/tag/0.10.20230714)

**Comment by sveeke** [comment]
> @sveeke this has already landed in version [`0.10.20230714`](https://github.com/BastilleBSD/bastille/releases/tag/0.10.20230714)

I see, great! Then it's only a matter of time before this version is released in the FreeBSD package.

**Comment by bmac2** [comment]
should be in last release.  Merged to main so closing as finished.
